### DISCUSSION TITLE
How to bind event handler with `passive` option set to `true`?

### DISCUSSION THREAD

**Discussion by zhaoyao91** [solidjs/solid#341] [Q&A]
Currently I have to get the ref then bind the event manually with options `{passive: true}`. Is there a way to bind event with options directly in view?

**Comment by ryansolid** [comment]
No.. this has been a challenge for all JSX frameworks, unfortunately. This is highlighted here: https://github.com/preactjs/preact/issues/428. There isn't particularly good syntax to handle this. Maybe an object syntax? Honestly, I wasn't too sure what to do here and was expecting some of library to make the move first. But it's been years and still nothing.

**Reply by Myzel394** [reply]
I think a good syntax would be to use Vue's syntax for this. Example

```jsx
<MyComponent
    onscroll:passive={console.log}
/>
```

**Comment by frenzzy** [comment]
You can do by adding a passive event listener to a DOM node via [`EventTarget.addEventListener()`](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener), for example:
```js
function Component(props) {
  let el
  const listener = (event) => console.log(event, props.children)
  createEffect(() => el.addEventListener('pointermove', listener, { passive: true }))
  onCleanup(() => el.removeEventListener('pointermove', listener))
  return <div ref={el}>{props.children}</div>;
}
```
Demo: https://playground.solidjs.com/?hash=517253560

**Reply by frenzzy** [reply]
What the use case to use passive event listener? I am asking because we usually use it for window object only but not for a specific DOM node. See example: https://playground.solidjs.com/?hash=-1932679897

<img src="https://user-images.githubusercontent.com/640669/107507591-dcef0480-6bb0-11eb-8edf-a2976fd461c2.gif" alt="demo" width="111"/>

**Reply by zhaoyao91** [reply]
We made some scroll container which hold some content inside and allow scrolling, and use a scroll listener on it to check if it is scrolled at bottom (we have other logic for this status).

**Reply by frenzzy** [reply]
I just thinking out loud, maybe the better solution for your case is using an [IntersectionObserver](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver) like described in this article: https://developers.google.com/web/updates/2017/09/sticky-headers
